### PR TITLE
Fix ScaleImg bug of assumption about IterationOrder

### PR DESCRIPTION
**Pull request by lnyng**
See #314 

The previous code generate the output image using the image factory of the input image. Since the output image has different dimension of the input image, the result could cause errors. For example, scaling up an large size image, so that the result image has more number of pixels than Integer.MAX_VALUE.

However, this case rarely happens in practice (scaling up an already large image?), and I find that no test on this feature could be run in reasonable amount of time. One significant reason is that, creating an `CellImg` of size 50000x50000 takes too much time (I never see it finish. Takes at least a few minutes), while in comparison, creating two `ArrayImg` of size 50000x25000 takes a few seconds. I think this is due to the use of default cell size in `DefaultCreateImgFactory`. I will file another issue and work on that later.

### CODE REVIEW

**Comment by dietzc** [comment]
what about using `Copy` here instead of hard coded loops?

**Comment by LeonYang5114** [comment]
@dietzc The branch is updated, and also include code that supports creating Cellimg. I think the `initialize` method I wrote in the `ScaleImg` op is not very nice, but I could not come up with a way to improve right now. Can you check? Thanks

File: src/main/java/net/imagej/ops/image/scale/ScaleImg.java [hunk 1]
```diff
@@ -52,56 +55,75 @@
...
+			new net.imglib2.realtransform.Scale(scaleFactors))), new FinalInterval(
+				newDims));
+
+		copyImg = (UnaryComputerOp) Computers.unary(ops(),
+			Ops.Copy.IterableInterval.class, Img.class, interval);
+	}
+
+	@Override
+	public Img<T> compute1(Img<T> input) {
+		final long[] newDims = new long[in().numDimensions()];
```
Comment (ctrueden) [inline src/main/java/net/imagej/ops/image/scale/ScaleImg.java]: Whoa. The `in()` method should never be called from `compute1`. Why are we not using `input` instead here?

**Comment by ctrueden** [comment]
`ScaleImg` became `DefaultScaleView`, so this bug is no longer relevant.

I almost cherry-picked the improvements to `DefaultCreateImgFactory`, but ultimately decided against it. We can add in the `targetCellSize` again when the need arises.

**Pull request closed without merging**